src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaContractTests.cs

```diff
@@ -0,0 +1,221 @@
+            AssertExtensions.Throws<ArgumentNullException>("pbeParameters", () => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>.Empty, null, Span<byte>.Empty, out _));
+        }
+
+        [Theory]
+        [MemberData(nameof(ArgumentValidationData))]
+        public static void ArgumentValidation(SlhDsaAlgorithm algorithm, bool shouldDispose)
+        {
+            using SlhDsa slhDsa = SlhDsaMockImplementation.Create(algorithm);
+
+            int publicKeySize = algorithm.PublicKeySizeInBytes;
+            int secretKeySize = algorithm.SecretKeySizeInBytes;
+            int signatureSize = algorithm.SignatureSizeInBytes;
+
+            if (shouldDispose)
+            {
+                // Test that argument validation exceptions take precedence over ObjectDisposedException
+                slhDsa.Dispose();
+            }
+
+            AssertExtensions.Throws<ArgumentException>("destination", () => slhDsa.ExportSlhDsaPublicKey(new byte[publicKeySize - 1]));
+            AssertExtensions.Throws<ArgumentException>("destination", () => slhDsa.ExportSlhDsaSecretKey(new byte[secretKeySize - 1]));
+            AssertExtensions.Throws<ArgumentException>("destination", () => slhDsa.SignData(ReadOnlySpan<byte>.Empty, new byte[signatureSize - 1], ReadOnlySpan<byte>.Empty));
+
+            // Context length must be less than 256
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("context", () => slhDsa.SignData(ReadOnlySpan<byte>.Empty, Span<byte>.Empty, new byte[256]));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("context", () => slhDsa.VerifyData(ReadOnlySpan<byte>.Empty, Span<byte>.Empty, new byte[256]));
+        }
+
+        public static IEnumerable<object[]> ApiWithDestinationSpanTestData =>
+            from algorithm in SlhDsaTestData.AlgorithmsRaw
+            from destinationLargerThanRequired in new[] { true, false }
+            select new object[] { algorithm, destinationLargerThanRequired };
+
+        private const int PaddingSize = 10;
+
+        [Theory]
+        [MemberData(nameof(ApiWithDestinationSpanTestData))]
+        public static void ExportSlhDsaPublicKey_CallsCore(SlhDsaAlgorithm algorithm, bool destinationLargerThanRequired)
+        {
+            using SlhDsaMockImplementation slhDsa = SlhDsaMockImplementation.Create(algorithm);
+
+            int publicKeySize = algorithm.PublicKeySizeInBytes;
+            byte[] publicKey = CreatePaddedFilledArray(publicKeySize, 42);
+
+            // Extra bytes in destination buffer should not be touched
+            int extraBytes = destinationLargerThanRequired ? PaddingSize / 2 : 0;
+            Memory<byte> destination = publicKey.AsMemory(PaddingSize, publicKeySize + extraBytes);
+
+            slhDsa.ExportSlhDsaPublicKeyCoreHook = _ => { };
+            slhDsa.AddDestinationBufferIsSameAssertion(destination[..publicKeySize]);
+            slhDsa.AddFillDestination(1);
```

Comment:
Previously this test only hooked up the two methods it cared about.  While the "AddDestinationBufferIsSameAssertion" helper should be failing if the wrong one gets called (because no two of our data formats have the same length), this particular test is claiming to assert that ExportSlhDsaPublicKey calls ExportSlhDsaPublicKeyCore, and the other ExportFooCore methods having a non-throw body means they now could be getting called unexpectedly.

Maybe it doesn't matter, but it feels worth pointing out that this change is adding laxity.

Reply:
Ah, right.  On my first read I had red it as "set" and was saving up the comment of "it's called Add but it's really Set", and then later realized it was daisy-chaining, so was actually "Add".  But the consequence thereof didn't quite click hard enough to reset this thought.